ghost/milestone-emails/lib/MilestonesEmailService.js

```diff
@@ -147,17 +152,19 @@ module.exports = class MilestonesEmailService {
         const shouldSendEmail = await this.#shouldSendEmail();
 
         if (shouldSendEmail) {
-            // TODO: hook up Ghostmailer or use StaffService and trigger event to send email
-            // await this.#mailer.send({
-            //     subject: 'Test',
-            //     html: '<div>Milestone achieved</div>',
-            //     to: 'test@example.com'
-            // });
-
             milestone.emailSentAt = new Date();
         }
 
-        return await this.#createMilestone(milestone);
+        const newMilestone = await this.#createMilestone(milestone);
+
+        if (this.#slackEnabled) {
+            console.log('🤖 → #saveMileStoneAndSendEmail → this.#slackEnabled', this.#slackEnabled);
+            console.log('🤖 → #saveMileStoneAndSendEmail → newMilestone.toJSON()', newMilestone.toJSON());
+
+            this.#events.emit(`milestone.${milestone.type}.achieved`, newMilestone.toJSON());
```

Comment:
If we switch to `@tryghost/domain-events` we can handle the events as part of the Milestone entity